File: ECommerceAPI/ECommerceAPI/Models/DocumentDetail.cs
using ECommerceAPI.Base.Attributes;
using System.ComponentModel.DataAnnotations;

namespace ECommerceAPI.Models
{
    public class DocumentDetail
    {
        public string DocumentType { get; set; }
        public int? DocumentDetailId { get; set; }
        public int? DocumentId { get; set; }
        public int? ProductId { get; set; }
        [ConditionalRequired("DocumentType=bill", "quantity_required")]
        public int? Quantity { get; set; }
        //[ChildValidation(new string[] { "Code;vat_required" })]
        [ChildValidation(new string[] { "Code;vat_required;DocumentType=bill" })]
        public Vat Vat { get; set; }
        [Required(ErrorMessage = "price_required")]
        public decimal? Price { get; set; }
        [Required(ErrorMessage = "discount_required")]
        public decimal? Discount { get; set; }
        public decimal? PriceWithDiscount { get; set; }
        [ChildValidation(new string[] { "Code;product_required" })]
        public Product Product { get; set; }
        public decimal? Sum { get; set; }
        public int VatId { get; set; }
    }
}

File: ECommerceFront/src/app/models/documentDetail.ts
import { Product } from './product';
import { Vat } from './vat';

export class DocumentDetail {
    documentType: string;
    documentDetailId: number;
    product: Product;
    quantity: number;
    vat: Vat;
    price: number;
    discount: number;
    priceWithDiscount: number;
    sum: number;
    constructor() {
        this.documentDetailId = null;
        this.product = new Product();
        this.quantity = null;
        this.vat = new Vat();
        this.price = null;
        this.discount = null;
        this.priceWithDiscount = null;
        this.sum = null;
        this.documentType = null;
    }
    assignObject(init?: Partial<DocumentDetail>) {
        Object.assign(this, init);
    }
}
